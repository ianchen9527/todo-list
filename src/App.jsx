import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.scss';
import routes from './constants/routes';
import Home from './components/pages/Home';
import Notes from './components/pages/Notes';
import Note from './components/pages/Note';
import NoteEdit from './components/pages/NoteEdit';

function App() {
  return (
    <Switch>
      <Route path={routes.HOME()} exact component={Home} />
      <Route path={routes.NOTES()} exact component={Notes} />
      <Route path={routes.NOTE()} exact component={Note} />
      <Route path={routes.NOTE_EDIT()} exact component={NoteEdit} />
    </Switch>
  );
}

export default App;
