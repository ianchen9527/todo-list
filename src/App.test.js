import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
