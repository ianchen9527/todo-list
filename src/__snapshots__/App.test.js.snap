// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/notes"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/notes/:id"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/notes/:id/edit"
  />
</Switch>
`;
